hwangsae/transmuxer.c

```diff
@@ -206,6 +208,43 @@ hwangsae_transmuxer_clear (HwangsaeTransmuxer * self)
   gst_clear_object (&priv->pipeline);
 }
 
+static gboolean
+_check_time_overlap (GSList * input_files)
```

Comment:
I guess this check could get simplified:
* add `guint64 duration` to `Segment` structure
* calculate `duration` from start and end times in `hwangsae_transmuxer_parse_segments()`
* then `segments` in `HwangsaeTransmuxerPrivate` will hold the list of input files ordered by start time
* now detect overlaps by iterating `segments` and checking that `base_time + duration < next_segment_base_time`

Reply:
Thanks for the suggestion. I can move the check to `hwangsae_transmuxer_parse_segments`, but  also 
- move `hwangsae_transmuxer_parse_segments` to the beginning of perform this check earlier.
- I don't see the reason to calculate `duration` as you need segment_end_time which is already and output of the `hwangsae_common_parse_times_from_filename`.

Do you agree?

Reply:
> move `hwangsae_transmuxer_parse_segments` to the beginning of perform this check earlier.

Yes, nothing against that.

> you need segment_end_time which is already and output of the `hwangsae_common_parse_times_from_filename`.

Yeah that should work too.